pkg/operator/sync_common.go

```diff
@@ -28,145 +28,195 @@ func manageControllerNS(c serviceCAOperator) (bool, error) {
 	if err != nil {
-		return nil, false, err
+		return err
+	}
+	_, mod, err := resourceapply.ApplyConfigMap(client, eventRecorder, requiredConfigMap)
+	if err != nil {
+		return err
 	}
-	return resourceapply.ApplyConfigMap(client, eventRecorder, requiredConfigMap)
+	*modified = *modified || mod
+
+	return nil
 }
 
-func manageSignerControllerDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, forceDeployment bool) (*appsv1.Deployment, bool, error) {
-	return manageDeployment(client, eventRecorder, options, "v4.0.0/service-serving-cert-signer-controller/", forceDeployment)
+func manageSignerControllerDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, modified *bool) error {
+	return manageDeployment(client, eventRecorder, options, "v4.0.0/service-serving-cert-signer-controller/", modified)
 }
 
-func manageAPIServiceControllerDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, forceDeployment bool) (*appsv1.Deployment, bool, error) {
-	return manageDeployment(client, eventRecorder, options, "v4.0.0/apiservice-cabundle-controller/", forceDeployment)
+func manageAPIServiceControllerDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, modified *bool) error {
+	return manageDeployment(client, eventRecorder, options, "v4.0.0/apiservice-cabundle-controller/", modified)
 }
 
-func manageConfigMapCABundleControllerDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, forceDeployment bool) (*appsv1.Deployment, bool, error) {
-	return manageDeployment(client, eventRecorder, options, "v4.0.0/configmap-cabundle-controller/", forceDeployment)
+func manageConfigMapCABundleControllerDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, modified *bool) error {
+	return manageDeployment(client, eventRecorder, options, "v4.0.0/configmap-cabundle-controller/", modified)
 }
 
-func manageDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, resourcePath string, forceDeployment bool) (*appsv1.Deployment, bool, error) {
+func manageDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.Recorder, options *operatorv1.ServiceCA, resourcePath string, modified *bool) error {
 	required := resourceread.ReadDeploymentV1OrDie(v4_00_assets.MustAsset(resourcePath + "deployment.yaml"))
 	required.Spec.Template.Spec.Containers[0].Image = os.Getenv("CONTROLLER_IMAGE")
 	required.Spec.Template.Spec.Containers[0].Args = append(required.Spec.Template.Spec.Containers[0].Args, fmt.Sprintf("-v=%s", options.Spec.LogLevel))
+	gen := getGeneration(client, operatorclient.TargetNamespace, required.Name)
```

Comment:
You can use library-go to help you with this:
https://github.com/openshift/cluster-authentication-operator/blob/522c921df8db8fec15ed303b591a9f0db40cc42d/pkg/operator2/operator.go#L329-L344